polar-core/src/polar.rs

```diff
@@ -216,9 +217,25 @@ impl Polar {
                             kb.add_rule_type(rule_type);
                         }
                     }
-                    parser::Line::ResourceBlock(block) => {
-                        diagnostics.append(&mut block.add_to_kb(kb))
-                    }
+                    parser::Line::ResourceBlock {
+                        keyword,
+                        resource,
+                        productions,
+                    } => match resource_block_from_productions(keyword, resource, productions) {
+                        Ok(block) => {
+                            let errors = block
+                                .add_to_kb(kb)
+                                .into_iter()
+                                .map(|e| e.set_context(Some(source), None));
```

Comment:
This and the one below it is what I'm referring to in this comment: https://github.com/osohq/oso/pull/1321/files#r740529825